src/main/java/fi/digitraffic/tis/vaco/crypt/EncryptionService.java

```diff
@@ -75,4 +96,81 @@ public <T> T decrypt(String cypher, Class<T> clz) {
             throw new VacoException("Failed to deserialize decrypted result", e);
         }
     }
+
+    @Value("${vaco.encryptionKeys.credentials}")
+    private String keyId;
+
+    private static KmsAsyncClient kmsAsyncClient;
+
+    public byte[] encryptBlob(AuthenticationDetails details) {
+
+            SdkBytes myBytes = SdkBytes.fromUtf8String(details.toString());
+            EncryptRequest encryptRequest = EncryptRequest.builder()
+                .keyId(keyId)
+                .plaintext(myBytes)
+                .build();
+
+            try {
+                EncryptResponse response = getAsyncClient()
+                    .encrypt(encryptRequest)
+                    .toCompletableFuture()
+                    .get();
+
+                return response.ciphertextBlob().asByteArray();
+            } catch (InterruptedException | ExecutionException e) {
+                throw new RuntimeException("Error during encryption", e);
```

Comment:
Replace `RuntimeException` with any child exception of `VacoException`. There might not be one which applies perfectly, so you may need to create your own.